### ISSUE TITLE
Change the names of the Password Strength Level

### ISSUE THREAD

**Issue by fungc-io** [authgear/authgear-server#4225]
Our password strength level is conflicting with the description of the strength estimation library. 

* "Very Unguessable" refers to "Safely Unguessable"
* "Extremely Unguessable" refers to "Very Unguessable"

Suggest to change them to:

* Disabled
* Very Weak
* Weak
* Fair
* Strong
* Very Strong

It's also easier to understand for layman

<img src="https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/953ae84b-54dd-4751-a320-9b9d98e1d93f/ca3855f1-e7fb-406d-ba0e-d658f7e8ac90?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi85NTNhZTg0Yi01NGRkLTQ3NTEtYTMyMC05YjlkOThlMWQ5M2YvY2EzODU1ZjEtZTdmYi00MDZkLWJhMGUtZDY1OGY3ZThhYzkwIiwiaWF0IjoxNzE2MTkwNTMzLCJleHAiOjMzMjg2NzUwNTMzfQ.rCr3kLezBDFG-UpIOeHaRf356QxVNgS906ksMLFHf2s " alt="image.png" width="1604" height="1128" />

**Comment by chpapa** [comment]
Actually the name is not very understandable anyway.<br><br>What about if we rename it like these for our Authgear Portal?

* Disabled
* Very Weak: Guessable risky password (guesses < 10^3)
* Weak: Protection from throttled online attacks (guesses < 10^6)
* Fair: Protection from unthrottled online attacks (guesses < 10^8)
* Strong: Moderate protection from offline bruteforce on hashes (guesses <10^10)
* Very Strong: Strong protection from offline bruteforce on hashes (guesses >= 10^10)

Or something like that? to indicate both "no of guesses" and what type of attacks to expect protection from.

**Comment by fungc-io** [comment]
Yea, it will be useful to have a description. But it maybe too wordy in a dropdown

I'm thinking of presenting them with a vertical slider like this:

<img src="https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/2c662bf5-e63e-49e8-9fbe-fcd835308e50/f1c9f6ee-b171-4d64-8535-8786310d5f2a?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi8yYzY2MmJmNS1lNjNlLTQ5ZTgtOWZiZS1mY2Q4MzUzMDhlNTAvZjFjOWY2ZWUtYjE3MS00ZDY0LTg1MzUtODc4NjMxMGQ1ZjJhIiwiaWF0IjoxNzE2MTk1NjkyLCJleHAiOjMzMjg2NzU1NjkyfQ.Kqkw-6NSOUr_MJhjpnnKl2vhK4TmJ_Y8TkY8WNlzfmc " alt="image.png" width="388" height="441" />

**Comment by fungc-io** [comment]
Hi @arvinho,

This refers to the password strength here,

We are thinking to 1) change the naming, 2) add description to each option.

What to you think is a good way to present them?

<img src="https://uploads.linear.app/860e83ae-05f0-4c4b-bd93-8a4b8d251836/8972b96a-6cdb-4460-b183-afdedba655e5/058107cf-26bc-4192-81bf-4c80d04117c5?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzg2MGU4M2FlLTA1ZjAtNGM0Yi1iZDkzLThhNGI4ZDI1MTgzNi84OTcyYjk2YS02Y2RiLTQ0NjAtYjE4My1hZmRlZGJhNjU1ZTUvMDU4MTA3Y2YtMjZiYy00MTkyLTgxYmYtNGM4MGQwNDExN2M1IiwiaWF0IjoxNzE2MjgxMTE1LCJleHAiOjMzMjg2ODQxMTE1fQ.Yp1dRX9g1VIpZSoOoyOqtJMHctNW-ZcmSyR9nHvE2_s " alt="image.png" width="815" height="516" />

**Comment by chpapa** [comment]
@arvinhoux 

1. It's not "no password set", password is still required (default is min length = 8)
2. The description, while I agree is easier for people to understand, are not technically accurate. I'm afraid it would also create trouble for us when enterprise users review it.

2nd thoughts, we should actually mention the password strength level here is based on zxcvbn, and maybe have a link to our help doc to explain it better.

Maybe we just add on the UI it is based on zxcvbn, and use the official description and score to make it clearer?

**Comment by arvinhoux** [comment]
I misunderstood earlier. Adjusting it to "None" to represent not using it may be more direct. <br><br>Alternatively, we can consider including a reference table in our documentation mapping to the official zxcvbn explanations, and provide a link to the official source. The benefit of doing so is that it can aid general users' understanding while also avoiding potential issues arising from misunderstandings.

**Comment by fungc-io** [comment]
@arvinho, such table and details are already in the help doc: [https://docs.authgear.com/security/password-strength#how-password-strength-is-calculated-in-authgear](https://docs.authgear.com/security/password-strength#how-password-strength-is-calculated-in-authgear)

Maybe just add a paragraph/help text in the minimum strength selection in the Portal UI and link to the doc.

"Strength level based on \[zxcvbn\](link) algorithm"

Can you help design that?
